indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -124,7 +124,9 @@ public TaskStatus call()
                         final File attemptDir = new File(taskDir, attemptUUID);
 
                         final ProcessHolder processHolder;
-                        final int childPort = portFinder.findUnusedPort();
+                        final int childQueryPort = portFinder.findUnusedPort();
```

Comment:
It's probably OK for ForkingTaskRunner (aka middle manager) because it assumes it has complete control over the node. For certain other cluster execution services you'll want the ability to specify the port via java option without any requirement on sequential-ness.